src/token.rs

```diff
@@ -1,12 +1,12 @@
 use clap::ArgMatches;
 
-use crate::util::file::*;
+use crate::{error::AocError, util::file::*};
 
-pub async fn token(matches: &ArgMatches)
+pub async fn token(matches: &ArgMatches) -> Result<(), AocError>
 {
     if let Some(token) = matches.get_one::<String>("set")
     {
-        let mut path = cargo_path(std::env::current_dir().unwrap()).await.unwrap();
+        let mut path = cargo_path(std::env::current_dir()?).await?;
```

Comment:
While you're at the refactor. You might as well change the API to take in void, since passing the same argument is kind of redundant

Reply:
Yes, It is redundant to pass it in for now at least. Then we can maybe change back if we need to find the `cargo_root` of another directory than `current_dir`, but I don't see that happening now at least